### PR TITLE
ddl: init new struct `SchemaChangeEvent` to replace the `DDLEvent`

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: ref #55723 

Problem Summary:

### What changed and how does it work?
Init a new struct `SchemaChangeEvent` which will be used to replace the `DDLEvent` in the next few PR. The new struct 's  field is empty and we will fill it gradually.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 
  The newly added `GetType` method has no effect to current logic.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/util/schema_change_notifier.go
Comment (lance6716) [inline pkg/ddl/util/schema_change_notifier.go:1]: Not sure it worth a separate package. We can do it in future PR

**Review by lance6716 — APPROVED** [review]

File: pkg/ddl/util/schema_change_notifier.go
Comment (fzzf678) [reply pkg/ddl/util/schema_change_notifier.go:1]: Agree，

File: pkg/statistics/handle/ddl/ddl.go [hunk 1]
```diff
@@ -198,6 +198,10 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
 	case model.ActionFlashbackCluster:
 		return h.statsWriter.UpdateStatsVersion()
 	}
+
+	switch t.SchemaChangeEvent.GetType() {
```
Comment (lance6716) [inline pkg/statistics/handle/ddl/ddl.go:202]:
```suggestion
	//revive:disable:empty-block
	switch t.SchemaChangeEvent.GetType() {
```

File: pkg/ddl/util/schema_change_notifier.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import "github.com/pingcap/tidb/pkg/parser/model"
+
+// SchemaChangeEvent stands for a schema change event. DDL will generate one event or multiple event (only for multi-schema change DDL).
```
Comment (Rustin170506) [inline pkg/ddl/util/schema_change_notifier.go:19]:
```suggestion
// SchemaChangeEvent stands for a schema change event. DDL will generate one event or multiple events (only for multi-schema change DDL).
```

**Review by Rustin170506 — APPROVED** [review]
Please fill in your PR body and description. 

Approve in advance. After you fill in it. Feel free to unhold.

/hold

File: pkg/statistics/handle/util/ddl_event.go [hunk 3]
```diff
@@ -44,6 +45,9 @@ type DDLEvent struct {
 	// It applies when a table structure is being changed from partitioned to non-partitioned, or vice versa.
 	oldTableID int64
 	tp         model.ActionType
+
+	// todo: replace DDLEvent by SchemaChangeEvent gradually
+	SchemaChangeEvent util2.SchemaChangeEvent
```
Comment (fzzf678) [inline pkg/statistics/handle/util/ddl_event.go:50]:
```suggestion
	SchemaChangeEvent ddlutil.SchemaChangeEvent
```

**Review by zimulala — APPROVED** [review]

**Pull request merged**